src/extension/abs_send_time_extension/abs_send_time_extension_test.rs

```diff
@@ -1,105 +1,105 @@
-    let loc = FixedOffset::west(5 * 60 * 60); // UTC-5
-    let tests = vec![
-        (
-            loc.ymd(1985, 6, 23).and_hms_nano(4, 0, 0, 0),
-            0xa0c65b1000000000 as u64,
-        ),
-        (
-            loc.ymd(1999, 12, 31).and_hms_nano(23, 59, 59, 500000),
-            0xbc18084f0020c49b as u64,
-        ),
-        (
-            loc.ymd(2019, 3, 27).and_hms_nano(13, 39, 30, 8675309),
-            0xe04641e202388b88 as u64,
-        ),
-    ];
-
-    for (t, n) in &tests {
-        let ntp = unix2ntp(Duration::from_nanos(t.timestamp_nanos() as u64));
-        assert_eq!(ntp, *n, "unix2ntp error");
-    }
+#[cfg(test)]
+mod tests {
+    use chrono::FixedOffset;
+    use chrono::TimeZone;
+    use std::ops::Sub;
+    use std::time::Duration;
 
-    for (t, n) in &tests {
-        let output = ntp2unix(*n);
-        let input = Duration::from_nanos(t.timestamp_nanos() as u64);
-        let diff = input.sub(output).as_nanos() as i128;
-        if diff < -ABS_SEND_TIME_RESOLUTION || ABS_SEND_TIME_RESOLUTION < diff {
-            assert!(
-                false,
-                "Converted time.Time from NTP time differs, expected: {:?}, got: {:?}",
-                input, output,
-            );
-        }
-    }
+    use crate::{
+        errors::ExtensionError,
+        extension::abs_send_time_extension::{ntp2unix, unix2ntp, AbsSendTimeExtension},
+    };
 
-    Ok(())
-}
+    const ABS_SEND_TIME_RESOLUTION: i128 = 1000;
 
-#[test]
-fn test_abs_send_time_extension_roundtrip() -> Result<(), Error> {
-    let tests = vec![
-        AbsSendTimeExtension { timestamp: 123456 },
-        AbsSendTimeExtension { timestamp: 654321 },
-    ];
-
-    for test in &tests {
-        let mut raw: Vec<u8> = vec![];
-        {
-            let mut writer = BufWriter::<&mut Vec<u8>>::new(raw.as_mut());
-            test.marshal(&mut writer)?;
+    #[test]
+    fn test_ntp_conversion() {
+        let loc = FixedOffset::west(5 * 60 * 60); // UTC-5
+        let tests = vec![
+            (
+                loc.ymd(1985, 6, 23).and_hms_nano(4, 0, 0, 0),
+                0xa0c65b1000000000 as u64,
+            ),
+            (
+                loc.ymd(1999, 12, 31).and_hms_nano(23, 59, 59, 500000),
+                0xbc18084f0020c49b as u64,
+            ),
+            (
+                loc.ymd(2019, 3, 27).and_hms_nano(13, 39, 30, 8675309),
+                0xe04641e202388b88 as u64,
+            ),
+        ];
+
+        for (t, n) in &tests {
+            let ntp = unix2ntp(Duration::from_nanos(t.timestamp_nanos() as u64));
+            assert_eq!(ntp, *n, "unix2ntp error");
+        }
+
+        for (t, n) in &tests {
+            let output = ntp2unix(*n);
+            let input = Duration::from_nanos(t.timestamp_nanos() as u64);
+            let diff = input.sub(output).as_nanos() as i128;
+            if diff < -ABS_SEND_TIME_RESOLUTION || ABS_SEND_TIME_RESOLUTION < diff {
+                panic!(
```

Comment:
why use panic! in test? instead of assert!(false...?

Reply:
Fixed